lms/djangoapps/grades/tests/test_new.py

```diff
@@ -328,6 +328,23 @@ def test_update(self):
             grade = self.subsection_grade_factory.update(self.sequence)
         self.assert_grade(grade, 1, 2)
 
+    def test_write_only_if_engaged(self):
+        """
+        Test that scores are not persisted when a learner has
+        never attempted a problem.
```

Comment:
...but they are persisted when score_deleted is true.

Reply:
✔️